dbms/src/Flash/Coprocessor/DAGQueryBlockInterpreter.cpp

```diff
@@ -246,13 +272,23 @@ void DAGQueryBlockInterpreter::handleJoin(const tipb::Join & join, DAGPipeline &
     size_t max_block_size_for_cross_join = settings.max_block_size;
     fiu_do_on(FailPoints::minimum_block_size_for_cross_join, { max_block_size_for_cross_join = 1; });
 
+    size_t shuffle_partition_num = 0;
+    if (enableFineGrainedShuffle(fine_grained_shuffle_count))
+    {
+        auto receiver_source_num_info = getReceiverSourceNumInfo(*build_pipeline.firstStream());
+        RUNTIME_ASSERT(receiver_source_num_info.first == 1);
```

Comment:
Maybe `RUNTIME_CHECK` is better, to avoid tidb side problems causing tiflash crash.

Reply:
Yeah, you're right, missed this one, I'll change it.

Reply:
Done.